### PR TITLE
Update appearance of help button component

### PR DESCRIPTION
**Pull request by janfaracik**
This is a small PR to add the help button component mentioned in #6307.

image

### Proposed changelog entries

* Updated help button for some pages

### Changes for plugin developers

The appearance of the help button has changed (need to add to design library)

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@NotMyFault 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by daniel-beck** [comment]
Why are there 132 commits in this PR? Is this a side-effect of squash-merging since your branches are old? This doesn't exactly facilitate reviews.

File: war/src/main/less/form/layout.less
Comment (NotMyFault) [inline war/src/main/less/form/layout.less:99]:
```suggestion
    content: "";
```
I took care of that in a previous PR, yet it sneaked its way in again.

turns into

The theme plugins are unaffected by this change, it solely takes care of browser extensions like dark reader.

**Review by NotMyFault — COMMENTED** [review]
It feels a bit weird to have tooltips vanishing while you hover on the item still:

https://user-images.githubusercontent.com/13383509/159573287-bc924d4d-c448-41ab-bd56-c180ab5ad606.mp4

File: war/src/main/less/form/layout.less
Comment (janfaracik) [reply war/src/main/less/form/layout.less]: Ah blimey, no idea how that got back in there. I've removed it now and reworked slightly how the button works so it's consistent across browsers/OSs. Cheers 👍

**Comment by timja** [comment]
> Why are there 132 commits in this PR? Is this a side-effect of squash-merging since your branches are old? This doesn't exactly facilitate reviews.

I rewrote history using Option 2 from https://blog.oddbit.com/post/2019-06-17-avoid-rebase-hell-squashing-wi/

Which was very nice and easy

File: core/src/main/resources/lib/hudson/help.jelly
Comment (daniel-beck) [inline core/src/main/resources/lib/hudson/help.jelly:37]: I know it's not directly related, but not opening external links in a new tab seems user hostile (and if they're not external, why is it not a normal inline help link? -- projectRelationship-help probably should be updated to not be weird)

```suggestion
  <a href="${attrs.href}" class="jenkins-button" tooltip="${attrs.tooltip ?: '%Help'}" target="_blank" rel="noopener noreferrer">
```

Happy to file an independent PR though if this goes beyond scope.

Comment (janfaracik) [reply core/src/main/resources/lib/hudson/help.jelly:37]: That's a great suggestion - cheers!

**Comment by janfaracik** [comment]
> It feels a bit weird to have tooltips vanishing while you hover on the item still:
> 
>  2398d904769265112f2fffa04758eec9.mp4

Looks like an issue/maybe intentional design with YUI tooltips, the tooltips don't vanish on the Tippy.js rewrite of them.
I'll put it up soon for review:

https://user-images.githubusercontent.com/43062514/159897156-819fd686-23fb-44fc-bd8d-ee91ce18a716.mov

**Review by fqueiruga — APPROVED** [review]
This is a great change @janfaracik , as always!
I think it would help to have a few screen captures for how it will look in forms (I'm assuming it  changes the form links because `lib/form/helpLink.jelly` is the one used there IIRC.

**Comment by NotMyFault** [comment]
> > It feels a bit weird to have tooltips vanishing while you hover on the item still:
> > 2398d904769265112f2fffa04758eec9.mp4
> 
> Looks like an issue/maybe intentional design with YUI tooltips, the tooltips don't vanish on the Tippy.js rewrite of them. I'll put it up soon for review:
> 
>  tooltips.mov

Oh I see, yeah, tooltips keep up on the tippy js demo. We can dismiss my comment here, then 👍🏻

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
